nav2_docking/opennav_docking/src/simple_charging_dock.cpp

```diff
@@ -322,7 +342,85 @@ void SimpleChargingDock::jointStateCallback(const sensor_msgs::msg::JointState::
   is_stalled_ = (velocity < stall_velocity_threshold_) && (effort > stall_effort_threshold_);
 }
 
+void SimpleChargingDock::startDetection()
+{
+  // Skip if already starting or ON
+  if (detector_state_ != DetectorState::OFF) {return;}
+
+  // 1. Service START request
+  //    We do not wait for response: non-blocking trigger
+  if (detector_client_) {
+    if (!detector_client_->wait_for_service(
+          std::chrono::duration<double>(detector_service_timeout_)))
+    {
+      RCLCPP_WARN(node_->get_logger(),
+                  "Detector service '%s' unavailable (timeout %.1f s), "
+                  "continuing with subscription only",
+                  detector_service_name_.c_str(), detector_service_timeout_);
+    } else {
+      auto req = std::make_shared<std_srvs::srv::Trigger::Request>();
+      auto future = detector_client_->async_send_request(req);
+    }
+  }
+
+  // 2. Subscription toggle
+  //    Only subscribe once; will set state to ON on first message
+  if (subscribe_toggle_ && !detected_pose_sub_) {
+    detected_pose_sub_ = node_->create_subscription<geometry_msgs::msg::PoseStamped>(
+      "detected_dock_pose", rclcpp::SensorDataQoS(),
+      [this](const geometry_msgs::msg::PoseStamped::SharedPtr pose) {
+        detected_dock_pose_ = *pose;
+        detector_state_ = DetectorState::ON;
+      });
+  }
+
+  detector_state_ = DetectorState::WAITING_START;
+  RCLCPP_INFO(node_->get_logger(), "Detector START requested");
+}
+
+void SimpleChargingDock::stopDetection()
+{
+  // Skip if already OFF
+  if (detector_state_ == DetectorState::OFF) {return;}
+
+  // 1. Service STOP request
+  //    Fire-and-forget; detector should teardown itself
+  if (detector_client_ && detector_client_->service_is_ready()) {
```

Comment:
With the Nav2 Utils Service Client, you can just do a `detector_client_ && detector_client_->invoke()`. Make sure to use the API with the timeout, we cannot have indefinite spins. There has to be error handling somewhere on failures.